lndclient/router_client.go

```diff
@@ -140,6 +145,17 @@ func (r *routerClient) SendPayment(ctx context.Context,
 	if request.LastHopPubkey != nil {
 		rpcReq.LastHopPubkey = request.LastHopPubkey[:]
 	}
+	if request.KeySend {
+		var preimage lntypes.Preimage
+		if _, err := rand.Read(preimage[:]); err != nil {
+			return nil, nil, err
+		}
+
+		// Override the payment hash.
+		rpcReq.DestCustomRecords = make(map[uint64][]byte)
+		rpcReq.DestCustomRecords[record.KeySendType] = preimage[:]
+		request.PaymentHash = preimage.Hash()
```

Comment:
Assert that `PaymentHash` wasn't set, to catch unintended use?

Reply:
done